internal/registry/response/module_versions.go

```diff
@@ -20,9 +20,16 @@ type ModuleProviderVersions struct {
 // ModuleVersion is the output metadata for a given version needed by CLI to
 // resolve candidate versions to satisfy requirements.
 type ModuleVersion struct {
-	Version    string              `json:"version"`
-	Root       VersionSubmodule    `json:"root"`
-	Submodules []*VersionSubmodule `json:"submodules"`
+	Version     string              `json:"version"`
+	Root        VersionSubmodule    `json:"root"`
+	Submodules  []*VersionSubmodule `json:"submodules"`
+	Deprecation *Deprecation        `json:"deprecation"`
```

Comment:
CJ and I pondered for a bit during review whether we needed an explicit test case to ensure _registries that do not support deprecation_ continued to work as expected. But I think I've decided that that's fine, because the absence of the deprecation field in the deserialized json is coherent in both cases — that is, a module from a registry that doesn't support deprecation is always by definition _not deprecated,_ so it's always coherent to treat it as not deprecated. 👍🏼